app/components/graph-editor/drawable-interfaces.ts

```diff
@@ -396,3 +286,127 @@ export class DefaultEdge implements DrawableEdge {
         this.source = sourceNode;
     }
 }
+
+
+// Functions ///////////////////////////////////////////////////////////////////
+
+
+/**
+ * isDrawableEdge  
+ *   Typeguard for drawable edges.
+ */
+export function isDrawableEdge(obj: any): obj is DrawableEdge {
+    return isItThat(obj, EDGE_DEFAULTS);
+}
+
+/**
+ * isDrawableNode  
+ *   Typeguard for drawable nodes.
+ */
+export function isDrawableNode(obj: any): obj is DrawableNode {
+    return isItThat(obj, NODE_DEFAULTS);
+}
+
+/**
+ * isItThat  
+ *   Checks if `it` has the same members as `that`.
+ */
+function isItThat(it: any, that: any): boolean {
+    if (!it)
+        return false;
+    let result = true;
+    for (let p in that) {
+        result =
+            result &&
+            (p in it) &&
+            (typeof that[p] === typeof it[p]);
+        if (!result)
+            return false;
+    }
+    return result;
+}
+
+/**
+ * isEdgeOverlapped  
+ *   Checks if two edges are overlapping.  
+ *   It is assumed that the edge being checked does not have the same source
+ *   and destination nodes.
+ */
+export function isEdgeOverlapped(e: DrawableEdge, nodeEdges: Map<DrawableNode, Set<DrawableEdge>>): boolean {
```

Comment:
Combine the two sets of edges into one iterable since the two loops are otherwise the same. Another option would be to extract the predicate of the loop into a function.